python/test/test_api.py

```diff
@@ -0,0 +1,146 @@
+import asyncio
+import queue
+import time
+import unittest
+
+import cupy
+import numpy
+import pytest
+import tritonserver
+
+
+class TrtionServerAPITest(unittest.TestCase):
+    def test_not_started(self):
+        server = tritonserver.Server()
+        with self.assertRaises(tritonserver.InvalidArgumentError):
+            server.is_ready()
+
+    def test_gpu_memory(self):
+        server = tritonserver.Server(
+            model_repository="/workspace/models",
+            log_verbose=True,
+            log_error=True,
+        )
+
+        server.start(blocking=True)
+
+        test = server.get_model("test")
+        fp16_input = cupy.array([[5], [6], [7], [8]], dtype=numpy.float16)
+        responses = test.infer(inputs={"fp16_input": fp16_input}, request_id="1")
+
+        for response in responses:
+            print(response)
+
+        responses = server.models["test"].infer(
+            inputs={"fp16_input": fp16_input}, request_id="1"
+        )
+
+        for response in responses:
+            print(response)
```

Comment:
what if the response is on GPU? Is it still going to be returned in numpy?

Reply:
in GPU case it will be cupy (except for BYTES case which didn't have an easy way to represent that in cupy)

Reply:
Check response v.s. expected response? `print` doesn't really test anything programmatically.

Reply:
Same for other `print` statements..